moto/emr/responses.py

```diff
@@ -76,6 +76,8 @@ def add_instance_groups(self):
             item["instance_count"] = int(item["instance_count"])
             # Adding support to EbsConfiguration
             self._parse_ebs_configuration(item)
+            # Adding support for auto_scaling_policy
+            Unflattener.unflatten_complex_params(item, "auto_scaling_policy")
```

Comment:
A generic "unflattening" of flattened querystring params, a la the immediately preceding EbsConfig special case, but using a generic mechanism.

Reply:
Can you leave a note/todo above the `_parse_ebs_configuration method`, to indicate it can/should be rewritten using the same Unflattener?
I'll create an issue later on to record this rewrite as technical debt

Reply:
Do you object to me just making the change as part of this PR?  It's (almost) as simple as replacing the call to _parse_ebs_configuraiton with unflatten_complex_params(item, "ebs_configuration").

Almost, because the unflattener parses every attribute as a string, while _parse_ebs_configuration knows which are ints, and the template as written needs one of them to be.  Addition of the jinja2 int filter fixes that, though.

Reply:
That would be even better!

Reply:
Done.